### DISCUSSION TITLE
Warning about outdated Command Line Tools still persists after updating

### DISCUSSION THREAD

**Discussion by whitebear60** [Homebrew/discussions#2773] [Getting started] [answered]
Recently I've switched from macOS High Sierra to Mojave and, after that, updated Xcode. But, when I'm trying to install any formulae or cask, brew says that there is newer version of Command Line Tools available.

I tried updating via `softwareupdate --all --install`, and tried removing `/Library/Developer/CommandLineTools` folder, then reinstalling via `Xcode-select --install` and via manual downloading the suggested version 11.3.1 from Apple Developer site, and, after all, the warning still persists.

`Xcode-select --version` says that its version installed is 2354.

Also attaching `brew doctor` output, maybe it can help.

Not sure this should be opened here or in the Issues page, but, anyway, I think that this is unintended...

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/42472132/149195846-ca5491b4-457a-447d-9fc3-8931878bb0b1.png">

**Comment by carlocab** [comment, marked as answer]
Please [don't use screenshots](https://meta.stackoverflow.com/a/285557) when sharing the output of commands in your terminal. They're often difficult to read and impossible to reuse (e.g. when looking up error messages). Instead, copy the text here and wrap them in three back-ticks (```) on the line before and on the line after to format them properly.

What's the output of `brew config`?

**Reply by whitebear60** [reply]
Sorry for that.

`brew config` output:
```
HOMEBREW_VERSION: 3.3.10-21-g3ba6afb
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3ba6afb1d3b3272bc17bf8ff978e071cd35502b1
Last commit: 4 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: fb70b1db1e59d3f057572a210226ecd4aedc8bd0
Core tap last commit: 39 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: quad-core 64-bit kabylake
Clang: 11.0.0 build 1100
Git: 2.21.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
macOS: 10.14.6-x86_64
CLT: 10.3.0.0.1.1562985497
Xcode: 11.3.1
```

**Reply by carlocab** [reply]
Your CLT is outdated:
```
CLT: 10.3.0.0.1.1562985497
```
Remove it with
```
sudo rm -rf /Library/Developer/CommandLineTools
```
and download the `Command Line Tools for Xcode 11.3.1` from the link shown to you by `brew doctor`.

What's the output of
```
xcode-select -p
/Library/Developer/CommandLineTools/usr/bin/clang --version
```
?

**Reply by whitebear60** [reply]
```
/Applications/Xcode.app/Contents/Developer
Apple LLVM version 10.0.1 (clang-1001.0.46.4)
Target: x86_64-apple-darwin18.7.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

**Reply by whitebear60** [reply]
Xcode-select, for unknown reason, downloads CLT version 10... The problem is now resolved, thanks for your assistance

**Reply by carlocab** [reply]
```
Apple LLVM version 10.0.1 (clang-1001.0.46.4)
```
Yea, your CLT is definitely still out of date. You must've made a mistake trying to update it earlier. Definitely remove `/Library/Developer/CommandLineTools` before downloading the installer from Apple's website. Don't use `xcode-select --install`, it might be picking up the wrong version for you.

**Reply by whitebear60** [reply]
Unfortunately, it picks wrong version... Anyway, now everything is up-to-date and works as intended. Thanks!

**Reply by pjrobertson** [reply]
I was also having this issue. Downloading with xcode-select --install was not working. I instead downloaded the files directly from: https://developer.apple.com/download/all/?q=command%20line%20tools%2011.3

**Reply by FoxMcWeezer** [reply]
@pjrobertson's solution worked for me when I needed to use brew on a prerelease version of MacOS. The error was

```
Error: Your Command Line Tools are too outdated.
Update them from Software Update in System Settings.
```

and I did not want to update my OS to get the recent version of Command Line Tools.
